### PR TITLE
Fix settings layout and empty match layout

### PR DESCRIPTION
**Pull request by amycshuang**
## Overview

Fixed layout for the settings screen and the empty match screen. Disabled dark mode.

## Changes Made

- Removed pause pear feature
- Separated edit availabilities into two screens 
- Disabled dark mode
- Fixed layout on empty match screen

## Test Coverage

- manual testing with simulator 

## Screenshots 

<details>

  <summary>Screen Shot Name</summary>

  

Simulator Screen Shot - iPhone 8 - 2020-11-22 at 10 56 21
  

Simulator Screen Shot - iPhone 11 Pro - 2020-11-22 at 11 02 29

Simulator Screen Shot - iPhone 11 Pro - 2020-11-22 at 11 02 35

Simulator Screen Shot - iPhone 11 Pro - 2020-11-22 at 11 02 39

</details>

### CODE REVIEW

**Comment by lucyxubroad** [comment]
Attach your addressed issues in your PR next time! Like addresses #88 , #83 , #74, #82 . Would also love for PRs to be slightly smaller! :)

File: CoffeeChat/Controller/EditTimeAvailabilityViewController.swift [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+            if let day = daysDict[day] {
+                let isAvailable = availabilities[day] != nil
+                cell.updateBackgroundColor(isAvailable: isAvailable)
+            }
+            if day == selectedDay {
+                dayCollectionView.selectItem(at: indexPath, animated: false, scrollPosition: .left)
+                cell.isSelected = true
+            }
+            return cell
+        }
```
Comment (pnor) [inline CoffeeChat/Controller/EditTimeAvailabilityViewController.swift:247]: I'd put a newline between the end bracket and `let section...`, to make is visually clear that above we are handling the case for `dayCollectionView` and below we are handling the other case

File: CoffeeChat/Info.plist [hunk 2]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>UIUserInterfaceStyle</key>
+	<string>Light</string>
```
Comment (pnor) [inline CoffeeChat/Info.plist:6]: 🌞

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+//  CoffeeChat
+//
+//  Created by Amy Chin Siu Huang on 11/19/20.
+//  Copyright © 2020 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+// MARK: - Location UICollectionView Header
+private class LocationHeaderLabel: UICollectionReusableView {
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:12]: Maybe move this into another folder? I think we should have a `View` folder. And our naming convention would be something like `LocationHeaderLabelView`

**Review by pnor — APPROVED** [review]
Looks good, nice work!

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        label.font = ._16CircularStdBook
+        addSubview(label)
+        label.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+        }
+    }
+
+    required init?(coder: NSCoder) { fatalError("init(coder:) has not been implemented") }
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:25]: I don't personally prefer for this to be in line.

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+
+    func configure(with text: String) { label.text = text }
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:27]: I don't personally prefer for this to be in line.

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+// MARK: - Location UICollectionView Header
+private class LocationHeaderLabel: UICollectionReusableView {
+
+    private let label = UILabel()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        label.font = ._16CircularStdBook
+        addSubview(label)
+        label.snp.makeConstraints { make in
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:20]: Can you add a line break between `addSubview(label)` and `label.snp.makeConstraints`?

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+    private let scheduleLocationSubLabel = UILabel()
+
+    // MARK: - Location CollectionView Sections
+    private enum LocationSection {
+        case campus([String])
+        case ctown([String])
+    }
+
+    // MARK: - Location Data Vars
+    private let campusReuseId = "campusReuseIdentifier"
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:47]: For reuse identifiers, our convention now is that we declare them as static variables inside the collection view cell that we will be using! So for example, inside of `SchedulingPlaceCollectionViewCell` file, we say `static let campusReuseId = "campusReuseIdentifier"`. And then in this file, we refer to that variable as `SchedulingPlaceCollectionViewCell.campusReuseId`

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+        locationsCollectionView.backgroundColor = .clear
+        locationsCollectionView.layer.masksToBounds = false
+        locationsCollectionView.register(SchedulingPlaceCollectionViewCell.self, forCellWithReuseIdentifier: campusReuseId)
+        locationsCollectionView.register(SchedulingPlaceCollectionViewCell.self, forCellWithReuseIdentifier: ctownReuseId)
+        locationsCollectionView.register(LocationHeaderLabel.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: campusHeaderId)
+        locationsCollectionView.register(LocationHeaderLabel.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: ctownHeaderId)
+        locationsCollectionView.isScrollEnabled = false
+        view.addSubview(locationsCollectionView)
+
+        selectedCampusLocations = campusLocations.filter {savedLocations.contains($0)}
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:122]: `selectedCampusLocations = campusLocations.filter { savedLocations.contains($0) }`?

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+        selectedCtownLocations = ctownLocations.filter {savedLocations.contains($0)}
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:123]: `selectedCtownLocations = ctownLocations.filter { savedLocations.contains($0) }?`

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+        "Cafe Jennie",
+        "Gimme Coffee",
+        "Goldie's Cafe",
+        "Green Dragon",
+        "Libe Cafe",
+        "Mac's Cafe",
+        "Martha's Cafe",
+        "Mattin's Cafe",
+        "Temple of Zeus"
+    ]
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:73]: Add line break here?

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+        scheduleLocationSubLabel.snp.makeConstraints { make in
+            make.top.equalTo(scheduleLocationLabel.snp.bottom).offset(5)
+            make.centerX.equalTo(view.snp.centerX)
+        }
+
+        locationsCollectionView.snp.makeConstraints { make in
+            make.top.equalTo(scheduleLocationSubLabel.snp.bottom).offset(5)
+            make.leading.trailing.equalTo(view.safeAreaLayoutGuide).inset(collectionViewSidePadding)
+            make.centerX.equalToSuperview()
+            make.bottom.equalTo(view.snp.bottom).offset(-collectionViewPadding)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:182]: `make.bottom.equalTo(view.snp.bottom).inset(collectionViewPadding)`

File: CoffeeChat/Controller/EditTimeAvailabilityViewController.swift [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+    }
+    
+    private var timeSections: [TimeSection] = []
+    
+    // MARK: - Time Data Vars
+    private let timeInteritemSpacing: CGFloat = 4
+    private let sectionInsets = UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 8)
+
+    private let dayCellReuseId = "dayCellReuseIdentifier"
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditTimeAvailabilityViewController.swift:53]: Same comment about reuse identifiers.

**Review by lucyxubroad — APPROVED** [review]

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+        }
+        backButton.addTarget(self, action: #selector(backPressed), for: .touchUpInside)
+        navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
+
+        saveBarButtonItem.title = "Save"
+        saveBarButtonItem.tintColor = .darkGreen
+        saveBarButtonItem.target = self
+        saveBarButtonItem.action = #selector(saveAvailability)
+        saveBarButtonItem.setTitleTextAttributes([
+            .font: UIFont.getFont(.medium, size: 20)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:151]: Just checking, is `UIFont.getFont(.medium, size: 20)` what we want?

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 14]
```diff
@@ -0,0 +1,296 @@
...
+    @objc private func backPressed() {
+        navigationController?.popViewController(animated: true)
+    }
+
+    @objc private func saveAvailability() {
+        // TODO: save new availability
+    }
+
+    private func setupConstraints() {
+        let collectionViewPadding = 30
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:165]: Maybe don't need a variable for this?

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 15]
```diff
@@ -0,0 +1,296 @@
...
+        let collectionViewSidePadding = 32
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:166]: Maybe no variable for this either.

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 16]
```diff
@@ -0,0 +1,296 @@
...
+
+        scheduleLocationLabel.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(20)
+            make.centerX.equalTo(view.snp.centerX)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:170]: Can't you do like `make.centerX.equalToSuperview()`

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift [hunk 17]
```diff
@@ -0,0 +1,296 @@
...
+        }
+
+        scheduleLocationSubLabel.snp.makeConstraints { make in
+            make.top.equalTo(scheduleLocationLabel.snp.bottom).offset(5)
+            make.centerX.equalTo(view.snp.centerX)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:175]: Can't you do like make.centerX.equalToSuperview()

File: CoffeeChat/Controller/EditTimeAvailabilityViewController.swift [hunk 18]
```diff
@@ -0,0 +1,320 @@
...
+    private func setupTimeSections() {
+        let morningSection = TimeSection(type: .morning, items: morningItems)
+        let afternoonSection = TimeSection(type: .afternoon, items: afternoonItems)
+        let eveningSection = TimeSection(type: .morning, items: eveningItems)
+        
+        timeSections = [morningSection, afternoonSection, eveningSection]
+    }
+    
+    private func setupConstraints() {
+        let dayCollectionViewWidth = daysAbbrev.count * 45
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditTimeAvailabilityViewController.swift:183]: Probably don't need this variable here.

File: CoffeeChat/Controller/EditTimeAvailabilityViewController.swift [hunk 19]
```diff
@@ -0,0 +1,320 @@
...
+
+        scheduleTimeLabel.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(20)
+            make.centerX.equalTo(view.snp.centerX)
+        }
+        
+        dayCollectionView.snp.makeConstraints { make in
+            make.top.equalTo(scheduleTimeLabel.snp.bottom).offset(20)
+            make.centerX.equalTo(view.snp.centerX)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditTimeAvailabilityViewController.swift:192]: make equal to superview?

File: CoffeeChat/Controller/EditTimeAvailabilityViewController.swift [hunk 20]
```diff
@@ -0,0 +1,320 @@
...
+        
+        timeSections = [morningSection, afternoonSection, eveningSection]
+    }
+    
+    private func setupConstraints() {
+        let dayCollectionViewWidth = daysAbbrev.count * 45
+
+        scheduleTimeLabel.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(20)
+            make.centerX.equalTo(view.snp.centerX)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditTimeAvailabilityViewController.swift:187]: make equal to superview?

File: CoffeeChat/Controller/EditTimeAvailabilityViewController.swift [hunk 21]
```diff
@@ -0,0 +1,320 @@
...
+        dayCollectionView.snp.makeConstraints { make in
+            make.top.equalTo(scheduleTimeLabel.snp.bottom).offset(20)
+            make.centerX.equalTo(view.snp.centerX)
+            make.height.equalTo(50)
+            make.width.equalTo(dayCollectionViewWidth)
+        }
+        
+        dayLabel.snp.makeConstraints { make in
+            make.top.equalTo(dayCollectionView.snp.bottom).offset(10)
+            make.centerX.equalTo(view.snp.centerX)
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditTimeAvailabilityViewController.swift:199]: superview?

File: CoffeeChat/Controller/EditLocationAvailabilityViewController.swift (hunk 13 above)
Comment (amycshuang) [reply CoffeeChat/Controller/EditLocationAvailabilityViewController.swift:151]: yes, the Figma has 20px medium font.

**Pull request merged**